tests/Feature/MemberControllerTest.php

```diff
@@ -0,0 +1,875 @@
+    {
+        $stub = public_path().'/img/nui.jpeg';
+        $name = str_random(8).'.jpeg';
+        $path = public_path().'/img/test/'.$name;
+        copy($stub, $path);
+        $file = new UploadedFile($path, $name, 'image/jpeg', filesize($path), null, true);
+        $array = [
+        'id' => 1,
+        'name' => 'thuuu',
+        'information' => 'interu',
+        'phone_number' => '09716945',
+        'birthday' => '1996-12-10',
+        'position_id' => 3,
+        'gender' => 'female',
+        'avatar' => $file,
+        ];
+        $array1 = Factory(Member::class)->create([
+        'name' => 't',
+        'information' => 'in',
+        'phone_number' => '0945',
+        'birthday' => '1996-12-08',
+        'position_id' => 1,
+        'gender' => 'male',
+        'avatar' => 'uht.png',
+        ]);
+        $response = $this->json('PUT', 'members/update', $array);
+        $response->assertStatus(422, $response->status());
+        $this->assertDatabaseMissing('members', [
+        'name' => $array['name'],
+        'information' => $array['information'],
+        'phone_number' => $array['phone_number'],
+        'birthday' => $array['birthday'],
+        'position_id' => $array['position_id'],
+        'gender' => $array['gender'],
+        ]);
+    }
+
+    public function testListMemberSuccess()
+    {
+        $project = factory(Member::class)->create();
+        $response = $this->get('members');
+        $response->assertStatus(200);
+    }
+
+    public function testDeleteMemberSuccess()
+    {
+        $array = Factory(Member::class)->create()->toArray();
+        $response = $this->call('DELETE', 'members/destroy', $array);
+        $this->assertEquals(200, $response->status());
+        $this->assertDatabaseMissing('members', [
+            'name' => $array['name'],
+            'information' => $array['information'],
+            'phone_number' => $array['phone_number'],
+            'birthday' => $array['birthday'],
+            'position_id' => $array['position_id'],
+            'gender' => $array['gender'],
+            'avatar' => $array['avatar']
+            ]);
+    }
+
+    public function testAddMemberWithNameRequire()
+    {
+        $array = [
+        'name' => '',
+        'information' => 'inter',
+        'phone_number' => '0978716945',
+        'birthday' => '1996-12-12',
+        'position_id' => 1,
+        'gender' => 'female',
+        'avatar' => '',
+        ];
+        $response = $this->json('POST', 'members/create', $array);
+        $response->assertStatus(422, $response->status());
+        $this->assertDatabaseMissing('members', [
+        'name' => $array['name'],
+        'information' => $array['information'],
+        'phone_number' => $array['phone_number'],
+        'birthday' => $array['birthday'],
+        'position_id' => $array['position_id'],
+        'gender' => $array['gender'],
+        ]);
+    }
+
+    public function testAddMemberWithValidName()
```

Comment:
The name of this test says the name is valid, but actually it is not valid.

Reply:
i have update name function